bigtable/client/filters_test.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/filters.h"
+
+#include <gmock/gmock.h>
+
+#include "bigtable/client/chrono_literals.h"
```

Comment:
What's the ordering of headers?

I was assuming it was:

1. Bigtable headers
2. Other third-party headers
3. C++ STL

However, here there's another Bigtable header at the end.

Reply:
The Google Style Guide strangely puts your own project headers after the system headers **except** for the current file header (and for tests the file associated with the tests).

I do not like it either, and willing to push for a change in the style guide (it is not what Lakos recommends for example).

You can teach clang-format to follow a different order too.